packages/flame/lib/src/collisions/broadphase/quadtree/has_quadtree_collision_detection.dart

```diff
@@ -66,28 +65,9 @@ mixin HasQuadTreeCollisionDetection on FlameGame
             (activeItemCenter.y - potentialCenter.y).abs() > minimumDistance!);
   }
 
-  bool onComponentTypeCheck(PositionComponent one, PositionComponent another) {
-    var checkParent = false;
-    if (one is GenericCollisionCallbacks) {
-      if (!(one as GenericCollisionCallbacks).onComponentTypeCheck(another)) {
-        return false;
-      }
-    } else {
-      checkParent = true;
-    }
-
-    if (another is GenericCollisionCallbacks) {
-      if (!(another as GenericCollisionCallbacks).onComponentTypeCheck(one)) {
-        return false;
-      }
-    } else {
-      checkParent = true;
-    }
-
-    if (checkParent && one is ShapeHitbox && another is ShapeHitbox) {
-      return onComponentTypeCheck(one.hitboxParent, another.hitboxParent);
-    }
-    return true;
+  bool onComponentTypeCheck(ShapeHitbox one, ShapeHitbox another) {
```

Comment:
I know this wasn't changed  in this PR but I would name these first and second, one hitbox1 and hitbox2.

Reply:
I can't choose one... but changing it to first and second.